src/main/webapp/app/course/course-scores/scores-storage.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { ScoresPerExerciseType } from 'app/entities/exercise.model';
+import { Result } from 'app/entities/result.model';
+import { CourseScores } from 'app/course/course-scores/course-scores';
+
+@Injectable({ providedIn: 'root' })
+export class ScoresStorageService {
+    private storedTotalScores: Map<number, CourseScores> = new Map();
+
+    private storedScoresPerExerciseType: Map<number, ScoresPerExerciseType> = new Map();
+
+    private participationResults: Result[] = [];
+
+    getStoredTotalScores(courseId: number): CourseScores | undefined {
+        return this.storedTotalScores.get(courseId);
+    }
+
+    setStoredTotalScores(courseId: number, totalScores: CourseScores): void {
+        this.storedTotalScores.set(courseId, totalScores);
+    }
+
+    getStoredScoresPerExerciseType(courseId: number): ScoresPerExerciseType | undefined {
+        return this.storedScoresPerExerciseType.get(courseId);
+    }
+
+    setStoredScoresPerExerciseType(courseId: number, scoresPerExerciseType: ScoresPerExerciseType): void {
+        this.storedScoresPerExerciseType.set(courseId, scoresPerExerciseType);
+    }
+
+    getStoredParticipationResult(participationId: number): Result | undefined {
+        return this.participationResults.find((result: Result) => result.participation?.id === participationId);
+    }
+
+    setStoredParticipationResults(participationResults: Result[] | undefined): void {
```

Comment:
At the moment this method is only called as a result of the `/for-dashboard` and the `{courseId}/for-dashboard` call which both happen as a result of loading the page (for the first time or reloading) and the array is thus initialized as empty. I changed it from `push` to just `this.participationResults = participationResults;` which makes it clearer the whole array is going to be replaced.

Reply:
Hmm, okay. I'd still suggest to use a map from ``result.participation.id --> result``here, for three reasons:

1. That would be more performant since you are iterating the entire array in ``getStoredParticipationResult`` to find a Result with a certain participation id

2. That way you can easily replace individual participation results as they come in, and don't have to throw away older ones. This could be beneficial in the future if you navigate between different courses as you don't loose results from other courses if you navigate to a page where the ``{courseId}/for-dashboard`` is exectued and the entire array gets replaced with just one course's results. Let's design the storage for participation results equal to how you did for total scores: You don't drop total scores of other courses there.

3. As far as I see, with your new solution, when you call the ``setStoredParticipationResults`` method from ``course-management.service.ts`` after the ``/for-dashboard`` call (the one for all courses) happened, you repeatedly call this method for each course, replacing the array each time, which doesn't make sense.